packages/lucia/src/middleware/index.ts

```diff
@@ -17,6 +17,7 @@ import type {
 	Response as ExpressResponse
 } from "express";
 import type { FastifyReply, FastifyRequest } from "fastify";
+import { H3Event, getCookie, getRequestURL, setCookie } from 'h3'
```

Comment:
We don't allow runtime dependencies. I'd rather not have users have issue with unrelated dependencies.

Reply:
would it be okay to add it as peer-dependency?

Reply:
I'm not sure how that improves it? Should've said "We discourage adding runtime dependencies for middlewares"

Reply:
Peer dependencies have to be installed manually, so users will not have issues with unrelated deps.

Reply:
I don't see how peer deps is related here? Either way, I just don't want to add dependencies just to get middlewares working.

Reply:
The point is that users of the h3 middleware already have h3 installed, and the package manager will just reuse this one when it  is declared as a peer dependency. Users not using h3, don't have to worry about this as the package manager will not install it. Alternatively, we could make each middleware its own package that properly declares dependencies etc.

I just don't know how to properly implement the h3 middleware without using these imports. That's just the way h3 provides access to cookies etc.

Reply:
> That's just the way h3 provides access to cookies etc.

That can't be true. You're passing the request and response object. Like I commented, if the current middleware is not working, please open a new issue with a reproduction.

Reply:
My concern is adding additional dependencies that we don't really need (e.g. the `setCookie()` method is pretty much the same as the one used in Node middleware), and making library/framework specific dependencies optional for users not using that framework. I don't think setting it as a peer dependency is helpful here.

Reply:
Should I then implement it as a new h3 middlware package?

Reply:
Like I commented, if the current middleware is not working, please open a new issue with a reproduction.

Reply:
It's sadly not easily reproducible as I only experienced the issues when deploying to azure. Together with the fix https://github.com/unjs/nitro/pull/1753 and this PR, it works well.

Either way, its only a matter of time until h3 will change their get/setCookies implementation since they want to remove the node dependency and instead use the events api (https://github.com/unjs/h3/issues/73). Actually, the missing implementation for cookies is the only reason why one cannot use this PR and deploy it on deno.

Reply:
Even without a reproduction, can you open a new bug report with some code examples?